mmaction/models/backbones/agcn_2s.py

```diff
@@ -0,0 +1,338 @@
+            , V)` format
+        - Output[1]: Graph adjacency matrix for output data in :math:`(K, V, V)
+            ` format
+
+        where
+            :math:`N` is a batch size,
+            :math:`K` is the spatial kernel size, as :math:`K == kernel_size[1]
+                `,
+            :math:`T_{in}/T_{out}` is a length of input/output sequence,
+            :math:`V` is the number of graph nodes.
+    """
+
+    def __init__(self,
+                 in_channels,
+                 out_channels,
+                 kernel_size,
+                 t_kernel_size=1,
+                 t_stride=1,
+                 t_padding=0,
+                 t_dilation=1,
+                 adj_len=17,
+                 bias=True):
+        super().__init__()
+
+        self.kernel_size = kernel_size
+
+        self.PA = nn.Parameter(torch.FloatTensor(3, adj_len, adj_len))
+        torch.nn.init.constant_(self.PA, 1e-6)
+
+        self.num_subset = 3
+        inter_channels = out_channels // 4
+        self.inter_c = inter_channels
+        self.conv_a = nn.ModuleList()
+        self.conv_b = nn.ModuleList()
+        self.conv_d = nn.ModuleList()
+        for i in range(self.num_subset):
+            self.conv_a.append(nn.Conv2d(in_channels, inter_channels, 1))
+            self.conv_b.append(nn.Conv2d(in_channels, inter_channels, 1))
+            self.conv_d.append(nn.Conv2d(in_channels, out_channels, 1))
+
+        if in_channels != out_channels:
+            self.down = nn.Sequential(
+                nn.Conv2d(in_channels, out_channels, 1),
+                nn.BatchNorm2d(out_channels))
+        else:
+            self.down = lambda x: x
+
+        self.bn = nn.BatchNorm2d(out_channels)
+        self.soft = nn.Softmax(-2)
+        self.relu = nn.ReLU()
+
+        for m in self.modules():
+            if isinstance(m, nn.Conv2d):
+                conv_init(m)
+            elif isinstance(m, nn.BatchNorm2d):
+                bn_init(m, 1)
+        bn_init(self.bn, 1e-6)
+        for i in range(self.num_subset):
+            conv_branch_init(self.conv_d[i], self.num_subset)
+
+    def forward(self, x, adj_mat):
+        """Defines the computation performed at every call."""
+        assert adj_mat.size(0) == self.kernel_size
+
+        N, C, T, V = x.size()
+        A = adj_mat + self.PA
+
+        y = None
+        for i in range(self.num_subset):
+            A1 = self.conv_a[i](x).permute(0, 3, 1, 2).contiguous().view(
+                N, V, self.inter_c * T)
+            A2 = self.conv_b[i](x).view(N, self.inter_c * T, V)
+            A1 = self.soft(torch.matmul(A1, A2) / A1.size(-1))  # N V V
+            A1 = A1 + A[i]
+            A2 = x.view(N, C * T, V)
+            z = self.conv_d[i](torch.matmul(A2, A1).view(N, C, T, V))
+            y = z + y if y is not None else z
+        y = self.bn(y)
+        y += self.down(x)
+
+        return self.relu(y), adj_mat
+
+
+@BACKBONES.register_module()
+class AGCN_2S(nn.Module):
```

Comment:
You can call this model AGCN. 2s mean 2stream, which means two-stream late fusion, not related to the AGCN model.